homeassistant/components/deconz/device_automation.py

```diff
@@ -0,0 +1,171 @@
+    CONF_DIM_UP_SHORT_RELEASE: 2002,
+    CONF_DIM_UP_LONG_PRESS: 2001,
+    CONF_DIM_UP_LONG_RELEASE: 2003,
+    CONF_DIM_DOWN_SHORT_RELEASE: 3002,
+    CONF_DIM_DOWN_LONG_PRESS: 3001,
+    CONF_DIM_DOWN_LONG_RELEASE: 3003,
+    CONF_LEFT_SHORT_RELEASE: 4002,
+    CONF_LEFT_LONG_PRESS: 4001,
+    CONF_LEFT_LONG_RELEASE: 4003,
+    CONF_RIGHT_SHORT_RELEASE: 5002,
+    CONF_RIGHT_LONG_PRESS: 5001,
+    CONF_RIGHT_LONG_RELEASE: 5003,
+}
+
+REMOTES = {"RWL021": HUE_DIMMER_REMOTE, "TRADFRI remote control": TRADFRI_REMOTE}
+
+TRIGGER_SCHEMA = vol.All(
+    vol.Schema(
+        {
+            vol.Required(CONF_DEVICE_ID): str,
+            vol.Required(CONF_DOMAIN): DOMAIN,
+            vol.Required(CONF_PLATFORM): "device",
+            vol.Required(CONF_TYPE): str,
+            vol.Required(CONF_UNIQUE_ID): str,
+        }
+    )
+)
+
+
+async def async_attach_trigger(hass, config, action, automation_info):
+    """Listen for state changes based on configuration."""
+    config = TRIGGER_SCHEMA(config)
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(config[CONF_DEVICE_ID])
+
+    trigger = REMOTES[device.model][config[CONF_TYPE]]
+
+    event_id = config[CONF_UNIQUE_ID]
+
+    state_config = {
+        event.CONF_EVENT_TYPE: CONF_DECONZ_EVENT,
+        event.CONF_EVENT_DATA: {CONF_UNIQUE_ID: event_id, CONF_EVENT: trigger},
+    }
+
+    return await event.async_trigger(hass, state_config, action, automation_info)
+
+
+async def async_trigger(hass, config, action, automation_info):
+    """Temporary so existing automation framework can be used for testing."""
+    return await async_attach_trigger(hass, config, action, automation_info)
+
+
+async def async_get_triggers(hass, device_id):
+    """List device triggers.
+
+    Make sure device is a supported remote model.
+    Retrieve the deconz event object matching device entry.
+    Generate device trigger list.
+    """
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(device_id)
+
+    if device.model not in REMOTES:
+        return
+
+    entry = hass.config_entries.async_get_entry(next(iter(device.config_entries)))
+    gateway = get_gateway_from_config_entry(hass, entry)
+
+    deconz_event = None
+    for item in gateway.events:
+
+        try:
+            if next(val for _, val in device.connections if val == item.serial):
+                deconz_event = item
+                break
+
+        except StopIteration:
+            continue
+
+    if deconz_event is None:
+        return
+
+    triggers = []
+
+    remote = REMOTES[device.model]
+
+    for trigger in remote.keys():
+        triggers.append(
+            {
+                CONF_DEVICE_ID: device_id,
+                CONF_DOMAIN: DOMAIN,
+                CONF_PLATFORM: "device",
+                CONF_TYPE: trigger,
+                CONF_UNIQUE_ID: deconz_event.serial,
```

Comment:
remind me again why we need unique ID here, and device ID is not enough?

Reply:
@balloob yesterday  Member
Why not use self._device.uniqueid ?
   @Kane610
Kane610 yesterday  Author Member
This is also well established and shouldn't be changed. Using unique id won't be very User friendly
   @balloob
balloob 23 hours ago  Member
but this means that the name changes, the automation breaks ?
   @balloob
balloob 23 hours ago  Member
Unique IDs should not be made for user friendliness, they should be unique and never change

balloobYesterday at 7:47 AM
add a new attribute to the event
"device_id"
or maybe add "unique_id"
so it's backwards compatible

Reply:
I thought we were talking about the event then, was I wrong? Why would this need to be included in the configuration too ?

Reply:
It is for the event

Reply:
Okay… but this is generating configurations for triggers. Why would we need the unique ID in there ? Can't we use the device ID and derive the serial from that?

Reply:
Sure, I'll move that logic

Reply:
It's important that we keep a single source of truth. Imagine the unique ID and the device becoming out of sync.